### ISSUE TITLE
Update dependencies

### ISSUE THREAD

**Issue by lavigne958** [burnash/gspread#1125]
## Overview

Update out of date dependencies.
notable dependency to update: `google-auth`. it does not seems to work anymore, we must upgrade to a more recent version without any breaking change.

**Comment by sam-s** [comment]
I still get
```
gspread 5.7.0 requires google-auth==1.12.0, but you have google-auth 2.14.1 which is incompatible.
gspread 5.7.0 requires google-auth-oauthlib==0.5.3, but you have google-auth-oauthlib 0.7.1 which is incompatible.
```

**Comment by lavigne958** [comment]
Hi yes, i messed up when I changed the dependencies of gspread. I set a fixed version of both dependencies.

I new fix is coming, on the next release: 5.7.1
See issue #1145

**Comment by lavigne958** [comment]
I just made the new release, you should now be able to upgrade gspread to latest version, which is 5.7.1, and everything should work fine.
